src/net.cpp

```diff
@@ -1857,7 +1891,10 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                 }
             } else {
                 // Not a feeler
-                std::tie(addr, addr_last_try) = addrman.Select();
+                // If preferred_net has a value set, pick an extra outbound
+                // peer from that network. The eviction logic in net_processing
```

Comment:
Do we have any risk of TOR and I2P peers bouncing each other forever?
Say, you have 7 ip4 peers and 1 Tor peer. I2P bounces Tor. In the next step, Tor (maybe another node, or maybe the same) bounces I2P node back.

Reply:
This example should be prevented by the "Protect extra full outbound peers by network" commit, shouldn't it? ie the process would be:

 * 7 ipv4 + 1 tor
 * 7 ipv4 + 1 tor + 1 i2p (via "network specific connection")
 * 6 ipv4 + 1 tor + 1 i2p (tor and i2p nodes protected from eviction due to `if (!m_connman.MultipleManualOrFullOutboundConns(pnode->addr.GetNetwork())) return;` code path)